packages/studio-web/src/app/editor/editor.component.ts

```diff
@@ -263,16 +263,20 @@ export class EditorComponent implements OnDestroy, OnInit, AfterViewInit {
     // Create missing body element
     const body = document.createElement("body");
     body.id = "t0b0";
-    const textNode = element.children[0];
-    if (textNode) {
-      while (textNode.hasChildNodes()) {
+    const textNode = element.querySelector("text");
+    if (textNode?.querySelector("body") == null) {
+      while (textNode?.hasChildNodes()) {
         // @ts-ignore
         body.appendChild(textNode.firstChild);
       }
-      textNode.appendChild(body);
-    }
+      textNode?.appendChild(body);
+    } /**/
```

Comment:
I have to say I don't understand how this code works, but I tested it carefully, and what I can see is that it does work. I took a readalong, changed its `body` tag to something else in the base64-encoded .readalong file, and this code now correctly insert a new `body` tag in the right place. Whereas the old code did not, and causes trouble as we know.